Lib/functools.py

```diff
@@ -193,6 +193,65 @@ def total_ordering(cls):
     return cls
 
 
+################################################################################
+### topological sort
+################################################################################
+
+class CycleError(Exception):
+    """Cycle Error"""
```

Comment:
This docstring could be improved, perhaps "Graph passed to topsort has cyclic dependency."